### DISCUSSION TITLE
Add email obfuscation?

### DISCUSSION THREAD

**Discussion by h-enk** [gethyas/hyas#192] [Ideas]
> I don't understand this one - the link render hook would do the same thing as the shortcode, whether or not obfuscation is effective, without having to use the shortcode. So better and easier workflow for editors. And markdown supports email links whether or not you're using a render hook, also making the shortcode unnecessary: `[email@domain.com](mailto:email@domain.com)` or `<email@domain.com>`

I'd prefer using render hooks (when possible) over shortcodes (non portability) — I think we both agree.

I'm just not sure about adding email obfuscation. It's _only_ making it harder for bots to scrape email addresses (I think it's better to avoid using email addresses), and what _are_ current best practices?

For now, I would say, if you would like to add email obfuscation, bring your own.

Aside: I like this approach (am using it on a couple of websites): [Adding simple email address obfuscation for your blog like Cloudflare Scrape Shield](https://andrewlock.net/simple-obfuscation-of-email-addresses-using-javascript/) — but it's more of a process (and wouldn't want to see it implemented in Hyas)

**Comment by timtorres** [comment]
Okay great – that's my point, to remove the email shortcode altogether. The only thing it's adding is the obfuscation, which is achievable through the render hook if wanted.

There are best practices for the link behaviour, which is to not alter the behaviour at all, but the _method_ varies. I am fine not including obfuscation but it is kind of nice to add some level of protection (what you have now in the shortcode) invisibly, whether or not the site author or editor chooses to display emails on their site. There are arguments for doing that instead of using a contact form.

**Reply by h-enk** [reply]
All right! See: https://github.com/h-enk/hyas/pull/195

**Reply by h-enk** [reply]
Thanks for your input — now part of Hyas [v1.5.3](https://github.com/h-enk/hyas/releases/tag/v1.5.3)
